Assets/Scripts/SS3D/Systems/Entities/Humanoid/Ragdoll.cs

```diff
@@ -0,0 +1,334 @@
+			ToggleKinematic(true);
+		}
+        
+		private void Update()
+		{
+            if (IsServer && _isKnockdownTimed && IsKnockedDown)
+            {
+                _knockdownTimer -= Time.deltaTime;
+                if (_knockdownTimer <= 0)
+                {
+                    Recover();
+                }
+            }
+            switch (_currentState)
+            {
+                case RagdollState.Walking:
+                    WalkingBehavior();
+                    break;
+                case RagdollState.Ragdoll:
+                    RagdollBehavior();
+                    break;
+                case RagdollState.BonesReset:
+                    BonesResetBehavior();
+                    break;
+                case RagdollState.StandingUp:
+                    StandingUpBehavior();
+                    break;
+            }
+        }
+
+        private void WalkingBehavior() { }
+        /// <summary>
+        /// Cast knockdown, that isn't going to expire until Recover()
+        /// </summary>
+        [ServerRpc(RequireOwnership = false)]
+        public void KnockdownTimeless()
+        {
+            _isKnockdownTimed = false;
+            IsKnockedDown = true;
+        }
+        /// <summary>
+        /// Knockdown the character for some time.
+        /// </summary>
+        /// <param name="seconds"></param>
+        [ServerRpc(RequireOwnership = false)]
+        public void Knockdown(float seconds)
+        {
+            _isKnockdownTimed = true;
+            _knockdownTimer += seconds;
+            IsKnockedDown = true;
+        }
+        private void Knockdown()
+        {
+            _currentState = RagdollState.Ragdoll;
+            ToggleController(false);
+            ToggleAnimator(false);
+            
+            if (!IsOwner) return;
+            Vector3 movement = _humanoidLivingController.TargetMovement * 3;
+            ToggleKinematic(false);
+            foreach (Transform part in _ragdollParts)
+            {
+                part.GetComponent<Rigidbody>().AddForce(movement, ForceMode.VelocityChange);
+            }
+        }
+
+        private void RagdollBehavior()
+        {
+            // Only the owner handles ragdoll's physics
+            if (!IsOwner) return;
+            AlignToHips();
+        }
+        /// <summary>
+        /// Adjust player's position and rotation
+        /// </summary>
+        private void AlignToHips()
+        {
+            IsFacingDown = _hips.transform.forward.y < 0;
+            Vector3 originalHipsPosition = _hips.position;
+            Vector3 newPosition = _hips.position;
+            // Get the lowest position
+            if (Physics.Raycast(transform.position, Vector3.down, out RaycastHit hitInfo))
+            {
+                // It's important to use + 0.0051f, because otherwise the player gets stuck
+                newPosition.y = hitInfo.point.y + 0.0051f;
```

Comment:
make that a private constant with an explicit name and a bit more doc regarding why this strange value, and where or why the player get stuck without it.